settings_test.go

```diff
@@ -177,4 +180,72 @@ func TestNewSettingsPartialFieldsOnlyFromValidationReq(t *testing.T) {
+		validationReq := &kubewarden_protocol.ValidationRequest{
+			Settings: []byte(`{"memory":{"maxLimit": "3G","defaultRequest": "1G", "defaultLimit": "2G", "ignoreValues": true}}`),
+		}
+		settings, err := NewSettingsFromValidationReq(validationReq)
+		if err != nil {
+			t.Fatalf("Unexpected error %+v", err)
+		}
+		if settings.Cpu != nil {
+			t.Fatal("cpu settings should be null")
+		}
+		expectedMemoryValue := resource.MustParse("3G")
+		if !settings.Memory.MaxLimit.Equal(expectedMemoryValue) {
+			t.Errorf("invalid memory max limit quantity parsed. Expected %+v, go %+v", expectedMemoryValue, settings.Memory.MaxLimit)
+		}
+		expectedMemoryValue = resource.MustParse("2G")
+		if !settings.Memory.DefaultLimit.Equal(expectedMemoryValue) {
+			t.Errorf("invalid memory default limit quantity parsed. Expected %+v, go %+v", expectedMemoryValue, settings.Memory.DefaultLimit)
+		}
+		expectedMemoryValue = resource.MustParse("1G")
+		if !settings.Memory.DefaultRequest.Equal(expectedMemoryValue) {
+			t.Errorf("invalid memory default request quantity parsed. Expected %+v, go %+v", expectedMemoryValue, settings.Memory.DefaultRequest)
+		}
+		if !settings.Memory.IgnoreValues {
+			t.Errorf("ignoreValues should be true")
+		}
+	})
+	t.Run("only cpu fields with ignoreValues", func(t *testing.T) {
+		validationReq := &kubewarden_protocol.ValidationRequest{
+			Settings: []byte(`{"cpu":{"maxLimit": "1","defaultRequest": "1", "defaultLimit": "1", "ignoreValues": true}}`),
+		}
+		settings, err := NewSettingsFromValidationReq(validationReq)
+		if err != nil {
+			t.Fatalf("Unexpected error %+v", err)
+		}
+		if settings.Memory != nil {
+			t.Fatal("memory settings should be null")
+		}
+		expectedCpuValue := resource.MustParse("1")
+		if !settings.Cpu.MaxLimit.Equal(expectedCpuValue) {
+
+			t.Errorf("invalid memory max limit quantity parsed. Expected %+v, go %+v", expectedCpuValue, settings.Cpu.MaxLimit)
+		}
+		if !settings.Cpu.DefaultLimit.Equal(expectedCpuValue) {
+			t.Errorf("invalid memory default limit quantity parsed. Expected %+v, go %+v", expectedCpuValue, settings.Cpu.DefaultLimit)
+		}
+		if !settings.Cpu.DefaultRequest.Equal(expectedCpuValue) {
+			t.Errorf("invalid memory default request quantity parsed. Expected %+v, go %+v", expectedCpuValue, settings.Cpu.DefaultRequest)
+		}
+		if !settings.Cpu.IgnoreValues {
+			t.Errorf("ignoreValues should be true")
+		}
+	})
+	t.Run("both cpu and memory fields with ignoreValues", func(t *testing.T) {
+		validationReq := &kubewarden_protocol.ValidationRequest{
+			Settings: []byte(`{"cpu":{"ignoreValues": true}, "memory":{"ignoreValues": true}}`),
+		}
+		settings, err := NewSettingsFromValidationReq(validationReq)
+		if err != nil {
+			t.Fatalf("Unexpected error %+v", err)
+		}
+		if !settings.Cpu.IgnoreValues {
+			t.Errorf("ignoreValues should be true")
+		}
+		if !settings.Memory.IgnoreValues {
+			t.Errorf("ignoreValues should be true")
+		}
+	})
```

Comment:
Thanks for adding these tests! But I think we are starting to have too much duplicate code in the tests. Can you try to move the validation code into a function and call it passing the `Cpu` or `Memory` and the expected values. Something line this:

```
if err := checkSettingsValues(settings.Cpu, expectedMaxLimit, expectedDefaultLimit, expectedDefaultRequest, expectedIgnoreValues); err != nil {
    t.Errorf("invalid Cpu settings: %v", err)

}
```

Reply:
Is your goal to replace most setting test logic with this `checkSettingsValues` logic? Some tests are checking with `.MustParse()` and most are just comparing strings.

Reply:
This should be taken care of with https://github.com/kubewarden/container-resources-policy/pull/18/commits/9bfcf0f8bb6e9f20d1e52e97317a4813e9408944

Reply:
That's better! Thanks!